doc/source/programs/gdal_vector_select.rst

```diff
@@ -0,0 +1,132 @@
+      --config <KEY>=<VALUE>                               Configuration option [may be repeated]
+      --progress                                           Display progress bar
+
+    Options:
+      -l, --layer, --input-layer <INPUT-LAYER>             Input layer name(s) [may be repeated]
+      -f, --of, --format, --output-format <OUTPUT-FORMAT>  Output format ("stream" allowed)
+      --co, --creation-option <KEY>=<VALUE>                Creation option [may be repeated]
+      --lco, --layer-creation-option <KEY>=<VALUE>         Layer creation option [may be repeated]
+      --overwrite                                          Whether overwriting existing output is allowed
+      --update                                             Whether to open existing dataset in update mode
+      --overwrite-layer                                    Whether overwriting existing layer is allowed
+      --append                                             Whether appending to existing layer is allowed
+      --output-layer <OUTPUT-LAYER>                        Output layer name
+      --exclude                                            Exclude specified fields
+                                                           Mutually exclusive with --ignore-missing-fields
+      --ignore-missing-fields                              Ignore missing fields
+                                                           Mutually exclusive with --exclude
+
+    Advanced Options:
+      --if, --input-format <INPUT-FORMAT>                  Input formats [may be repeated]
+      --oo, --open-option <KEY=VALUE>                      Open options [may be repeated]
+
+
+Description
+-----------
+
+:program:`gdal vector select` can be used to select a subset of fields.
+
+``select`` can also be used as a step of :ref:`gdal_vector_pipeline_subcommand`.
+
+Standard options
+++++++++++++++++
+
+.. include:: gdal_options/of_vector.rst
+
+.. include:: gdal_options/co_vector.rst
+
+.. include:: gdal_options/overwrite.rst
+
+.. option:: --fields <FIELDS>
+
+    Comma-separated list of fields from input layer to copy to the new layer
+    (or to exclude if :option:`--exclude` is specified)
+
+    Field names with spaces, commas or double-quote
+    should be surrounded with a starting and ending double-quote character, and
+    double-quote characters in a field name should be escaped with backslash.
+
+    Depending on the shell used, this might require further quoting. For example,
+    to select ``regular_field``, ``a_field_with space, and comma`` and
+    ``a field with " double quote`` with a Unix shell:
+
+    .. code-block:: bash
+
+        --fields "regular_field,\"a_field_with space, and comma\",\"a field with \\\" double quote\""
+
+    A field is only selected once, even if mentioned several times in the list.
+
+    Geometry fields can also be specified in the list. If the source layer has
+    no explicit name for the geometry field, ``_ogr_geometry_`` must be used to
```

Comment:
I think needed to include `_ogr_geometry_` explicitly might be surprising for users. I can imagine including it by default unless an `--exclude-geom` argument is passed. Would be curious what others think.

Reply:
FWIW, the R [sf](https://github.com/r-spatial/sf) package behaves this way (some discussion at https://github.com/r-spatial/sf/issues/121).

Reply:
> I think needing to include `_ogr_geometry_` explicitly might be surprising for users. I can imagine including it by default unless an `--exclude-geom` argument is passed

that's a good question. I'm not sure either. One thing that is certain is that the current difference in behavior between OGRSQL where "SELECT" automatically selects the geometry field (unless excluded thanks to your recent work on EXCLUDE) compared to all other SQL dialects where the geometry field must be explicitly selected surprises users a lot.

Reply:
ticketed at https://github.com/OSGeo/gdal/issues/11839